File: test/v.3.00.00/Test..WebResource.WebApi/Dev.DevKit.WebResource/entities/Account.js
//@ts-check
///<reference path="Account.d.ts" />
"use strict";
var formAccount = (function () {
	"use strict";
	/** @type DevKit.FormAccount */
	var form = null;
	async function onLoad(executionContext) {
		form = new DevKit.FormAccount(executionContext)
		await TestWebApi();
	}
	async function onSave(executionContext) {
	}
	async function TestWebApi() {
		var fetchData = {
			accountid: form.EntityId
		};
		var fetchXml = `
<fetch version='1.0' output-format='xml-platform' mapping='logical' distinct='false'>
  <entity name='account'>
    <attribute name='name'/>
    <attribute name='accountnumber'/>
    <attribute name='telephone1'/>
    <attribute name='creditlimit'/>
	<attribute name='creditlimit_base'/>
    <order attribute='name' descending='false'/>
    <filter type='and'>
      <condition attribute='accountid' operator='eq' value='${fetchData.accountid}'/>
    </filter>
  </entity>
</fetch>
`;
		fetchXml = "?fetchXml=" + encodeURIComponent(fetchXml);
		var rows = await Xrm.WebApi.retrieveMultipleRecords("account", fetchXml);
		var accountApi = new DevKit.AccountApi(rows.entities[0]);
		var accountnumber = accountApi.AccountNumber;
		accountApi.Name = "ABCD";
		accountApi.CreditLimit = null;
		accountApi.CreditLimit = 123.455;
		accountApi.PrimaryContactId = "AAAA";
		var a = accountApi.CreatedOn_UtcDateAndTime;
		var formatName = accountApi.FormattedValue.Name;
		if (accountApi.AccountCategoryCode === OptionSet.Account.AccountCategoryCode.Preferred_Customer) {
		}

		debugger;
	}
	return {
		OnLoad: onLoad,
		OnSave: onSave
	};
})();

File: test/v.3.33.33/ProjectTemplate/04.Plugin Project/vs2019/Dev.DevKit.Plugin.Account/Account.cs
﻿using Microsoft.Xrm.Sdk;
using System;

namespace Dev.DevKit.PluginAccount
{
    public partial class Account
    {
        #region --- PROPERTIES ---

        //public string StringField { get { return GetAliasedValue<string>("aliased.field"); } }
        //public int? IntField { get { return GetAliasedValue<int?>("aliased.field"); } }
        //public bool? BooleanField { get { return GetAliasedValue<bool?>("aliased.field"); } }
        //public DateTime? DateTimeField { get { return GetAliasedValue<DateTime?>("aliased.field"); } }
        //public EntityReference LookupField { get { return GetAliasedValue<EntityReference>("aliased.field"); } }
        //public xxxOptionSets.xxx? OptionSetField { get { return (xxxOptionSets.xxx?)GetAliasedValue<OptionSetValue>("aliased.field")?.Value; } }
        //public decimal? MoneyField { get { return GetAliasedValue<Money>("aliased.field")?.Value; } }

        #endregion

        #region --- STATIC METHODS ---

        public static Account Read_Record(IOrganizationService serviceAdmin, IOrganizationService service, ITracingService tracing, Guid? recordId)
        {
            var fetchData = new
            {
                accountid = recordId ?? Guid.Empty
            };
            var fetchXml = $@"
<fetch version='1.0' output-format='xml-platform' mapping='logical' distinct='false'>
  <entity name='account'>
    <all-attributes/>
    <filter type='and'>
      <condition attribute='accountid' operator='eq' value='{fetchData.accountid}'/>
    </filter>
  </entity>
</fetch>
";
            var rows = serviceAdmin.RetrieveMultiple<Account>(fetchXml);
            if (rows.Count == 1) return rows[0];
            return new Account();
        }

        #endregion
    }
}